src/lib/homeobject_impl.hpp

```diff
@@ -18,15 +19,22 @@ namespace homeobject {
 inline bool operator<(ShardInfo const& lhs, ShardInfo const& rhs) { return lhs.id < rhs.id; }
 inline bool operator==(ShardInfo const& lhs, ShardInfo const& rhs) { return lhs.id == rhs.id; }
 
+static constexpr uint32_t init_crc32 = 0;
+
+struct PG {
+    explicit PG(PGInfo info) : pg_info(std::move(info)) {}
+    PGInfo pg_info;
+    uint64_t shard_sequence_num{0};
+    ShardInfoPtrList shards;
```

Comment:
Why a `list`? ShardInfo's are sortable and should be unique in a given Pg. Should we not retain this being a `set`? Maybe we need to revisit the public API if that's the reason to use a `list` here?

Reply:
shard info is sortable and we can use std::set here,  but I am not sure if it is needed to sort the shard info inside one pg. if we do not need sort, list is enough.

Reply:
if you want to get a specified shard info from the shard_id, you can use the _shard_map, no need to touch this structure

Reply:
The only time we touch this is when we move/delete a shard, so I'm okay with a list. It makes the list_shards() operation potentially simpler since the result is a list as well.